python/pyarrow/src_arrow/CMakeLists.txt

```diff
@@ -0,0 +1,573 @@
+  set(XSIMD_PREFIX "${ARROW_CPP_SOURCE_DIR}/${ARROW_BUILD_DIR}/xsimd_ep/src/xsimd_ep-install")
+  if(EXISTS ${XSIMD_PREFIX}/include)
+    set(XSIMD_INCLUDE_DIR "${XSIMD_PREFIX}/include")
+  else()
+    set(XSIMD_PREFIX "$ENV{CONDA_PREFIX}")
+    set(XSIMD_INCLUDE_DIR "${XSIMD_PREFIX}/include")
+  endif()
+
+  add_library(xsimd INTERFACE IMPORTED)
+  if(CMAKE_VERSION VERSION_LESS 3.11)
+    set_target_properties(xsimd PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+                                          "${XSIMD_INCLUDE_DIR}")
+  else()
+    target_include_directories(xsimd INTERFACE "${XSIMD_INCLUDE_DIR}")
+  endif()
+
+  list(APPEND ARROW_PYTHON_SHARED_PRIVATE_LINK_LIBS xsimd)
+  list(APPEND ARROW_PYTHON_STATIC_LINK_LIBS xsimd)
+endif()
+
+set(ARROW_PYTHON_INCLUDES ${NUMPY_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIRS})
+
+# Inlude macros needed to find and use add_arrow_lib function
+include(BuildUtils)
+include(CMakePackageConfigHelpers)
+
+# Set the output directory for cmake module
+# (CMAKE_INSTALL_PREFIX = python/build/dist! should be set in setup.py!)
+set(ARROW_CMAKE_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+add_arrow_lib(arrow_python
+              CMAKE_PACKAGE_NAME
+              ArrowPython
+              PKG_CONFIG_NAME
+              arrow-python
+              SOURCES
+              ${ARROW_PYTHON_SRCS}
+              PRECOMPILED_HEADERS
+              "$<$<COMPILE_LANGUAGE:CXX>:pch.h>"
+              OUTPUTS
+              ARROW_PYTHON_LIBRARIES
+              DEPENDENCIES
+              ${ARROW_PYTHON_DEPENDENCIES}
+              SHARED_LINK_FLAGS
+              ${ARROW_VERSION_SCRIPT_FLAGS}
+              SHARED_LINK_LIBS
+              ${ARROW_PYTHON_SHARED_LINK_LIBS}
+              SHARED_PRIVATE_LINK_LIBS
+              ${ARROW_PYTHON_SHARED_PRIVATE_LINK_LIBS}
+              STATIC_LINK_LIBS
+              ${ARROW_PYTHON_STATIC_LINK_LIBS}
+              EXTRA_INCLUDES
+              "${ARROW_PYTHON_INCLUDES}")
+
+add_dependencies(arrow_python ${ARROW_PYTHON_LIBRARIES})
+
+foreach(LIB_TARGET ${ARROW_PYTHON_LIBRARIES})
+  target_compile_definitions(${LIB_TARGET} PRIVATE ARROW_PYTHON_EXPORTING)
+endforeach()
+
+if(ARROW_BUILD_STATIC AND MSVC)
+  target_compile_definitions(arrow_python_static PUBLIC ARROW_STATIC)
+endif()
+
+if(ARROW_FLIGHT AND ARROW_BUILD_SHARED)
+
+  # Static gRPC means we cannot link to shared Arrow, since then
+  # we'll violate ODR for gRPC symbols
+  if(NOT ARROW_GRPC_USE_SHARED AND NOT ARROW_BUILD_STATIC)
+    message(FATAL_ERROR "Statically built gRPC requires ARROW_BUILD_STATIC=ON")
+  endif()
+
+  find_package(ArrowFlight REQUIRED)
+  find_package(gRPC CONFIG REQUIRED)
+  include_directories("${ARROW_CPP_SOURCE_DIR}/src" "${ARROW_CPP_SOURCE_DIR}/}/${ARROW_BUILD_DIR}/src")
+
+  # Using example
+  # https://github.com/apache/arrow/blob/7a0f00c16e084d194ae53d209b33b809cfc8f2d5/cpp/examples/arrow/CMakeLists.txt
+
+  if(ARROW_GRPC_USE_SHARED)
+    set(FLIGHT_EXAMPLES_LINK_LIBS arrow_flight_shared)
```

Comment:
this should probably be renamed something like `FLIGHT_LINK_LIBS` since this isn't an example